index.d.ts

```diff
@@ -746,6 +745,25 @@ declare module 'fastest-validator' {
 		messages?: MessagesType,
 	};
 
+	interface CompilationRule {
+		index: number;
+		ruleFunction: CompilationFunction
+		schema: ValidationSchema;
+		messages: MessagesType;
+		customValidation: (value?: string) => string;
+	}
+
+	interface Context {
+		index: number;
+		rules: ValidationRuleObject[];
+		fn: Function[];
+		customs: { [ruleName: string]: { schema: RuleCustom, messages: MessagesType } }
+	}
+
+	type CheckerFunction = (value: unknown, schema: ValidationSchema, path: string, parent: object | null, context: Context) => true | ValidationError[];
```

Comment:
`CheckerFunction` should have a generic type for `value` parameter. 
```suggestion
	type CheckerFunction<T = unknown> = (value: T, schema: ValidationSchema, path: string, parent: object | null, context: Context) => true | ValidationError[];
```

Reply:
The same case as in the previous conversation

Reply:
Guess it can be resolved by that comment https://github.com/icebob/fastest-validator/pull/122#discussion_r401724595